<?xml version="1.0"?>
<launch>
  <arg name="world_name" default="corridor" />
  <arg name="gui" default="true" />
  <arg name="run_camera" default="false"/>
  <arg name="map_file"  default="corridor"/>

  <param name ="/use_sim_time" value="true"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find cb_gazebo)/worlds/$(arg world_name).world"/>
    <arg name="gui" value="$(arg gui)"/>
  </include>

  <!--  ******************* Maps ***********************  -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find cb_gazebo)/maps/$(arg map_file).yaml">
    <param name="frame_id" value="map"/>
  </node>

  <!-- BEGIN ROBOT-->

  <include file="$(find cb_gazebo)/launch/one_car.launch" >
    <arg name="init_pose" value="-x 2 -y 3 -z 0" />
    <arg name="robot_name" value="robot_0" />
  </include>

  <node name="teleop" pkg="cb_gazebo" type="teleop_twist_keyboard_carlike.py"/>
  <node name="gazebo_odom" pkg="cb_gazebo" type="gazebo_odometry.py" output="screen" />

  <!-- BEGIN Obstacles-->
  <group ns="robot_1">
    <include file="$(find cb_gazebo)/launch/one_obstacle.launch" >
      <arg name="robot_name" value="robot_1" />
      <arg name="init_pose" value="-x 4 -y 3.74 -z 0" />
    </include>
    <param name="vel_y" value="0.5" />  
    <param name="vel_x" value="0.4" />
    <param name="pos_lb_x" value="3.0" />
		<param name="pos_ub_x" value="5.0" />
		<param name="pos_lb_y" value="0.5" />
		<param name="pos_ub_y" value="2.5" />  
  </group>

  <group ns="robot_2">
    <include file="$(find cb_gazebo)/launch/one_obstacle.launch" >
      <arg name="robot_name" value="robot_2" />
      <arg name="init_pose" value="-x 5.5 -y 3.0 -z 0" />
    </include>
    <param name="vel_y" value="0.0" />  
    <param name="vel_x" value="0.0" />  
    <param name="pos_lb_x" value="4.75" />
		<param name="pos_ub_x" value="6.25" />
		<param name="pos_lb_y" value="3.5" />
		<param name="pos_ub_y" value="5.5" />
  </group>

  <!-- <group ns="robot_3">
    <include file="$(find cb_gazebo)/launch/one_obstacle.launch" >
      <arg name="robot_name" value="robot_3" />
      <arg name="init_pose" value="-x 7 -y 5.38 -z 0" />
    </include>
    <param name="vel_y" value="0.3" />  
    <param name="vel_x" value="0.4" />  
  </group>

  <group ns="robot_4">
    <include file="$(find cb_gazebo)/launch/one_obstacle.launch" >
      <arg name="robot_name" value="robot_4" />
      <arg name="init_pose" value="-x 8.5 -y 1.56 -z 0" />
    </include>
    <param name="vel_y" value="0.3" />  
    <param name="vel_x" value="0.4" />  
  </group> -->

  

</launch>
